app/assets/javascripts/map.js

```diff
@@ -0,0 +1,210 @@
+      polygon = new Polygon(route.polyline)
+      route.displayRoute(response, true)
+    }
+  });
+};
+
+
+Route.prototype.displayRoute = function(response, drawMe){
+  drawMe = typeof drawMe !== 'undefined' ? drawMe : true;
+
+  directionsDisplay.setDirections(response);
+
+  if (drawMe) {
+    drawer = new Drawer();
+    drawer.draw(polygon.geoJson);
+  };
+}
+
+
+//----polyline decoder-----
+
+var PolylineDecoder = function(response){}
+
+PolylineDecoder.prototype.decodePolyline = function(response){
+  var coord_array = polyline.decode(response.routes[0].overview_polyline);
+  return coord_array.map(function(coordinate) {
+    return [coordinate[1], coordinate[0]];
+  })
+}
+
+//-------Polygon!------------
+
+var Polygon = function(polyline){
+  this.geoJson = this.createGeoJsonFromPolyline(polyline)
+  this.searchWithin(this.geoJson)
+}
+
+Polygon.prototype.createGeoJsonFromPolyline = function(polyline) {
+  var line = {
+    type:"Feature",
+    geometry:{
+      type:"LineString",
+      coordinates: polyline
+     }
+   }
+
+  var polygon = turf.buffer(line, 25, 'miles')
+  return polygon
+}
+
+Polygon.prototype.searchWithin = function(polygon){
+  $.ajax({
+  url: '/maps/search',
+  type: 'post',
+  beforeSend: function(xhr) { xhr.setRequestHeader('X-CSRF-Token', $('meta[name=csrf-token]').attr('content'))},
+  contentType: "application/json",
+  dataType: "json",
+  data: JSON.stringify(polygon),
+})
+  .success(function(response){
+    console.log(response)
+    loadMarkers(response.attractions)
+  })
+}
+
+//---------Draw-er---------------------
+var Drawer = function(){}
+
+Drawer.prototype.createLatLongObjects = function(geoJsonObject){
+  var latLongArray = []
+  var coordArray = geoJsonObject.features[0].geometry.coordinates[0]
+
+  coordArray.forEach(function(coord){
+    latLongArray.push(new google.maps.LatLng(coord[1], coord[0]))
+    return latLongArray
+  })
+  return latLongArray
+}
+
+Drawer.prototype.draw = function(geoJsonObject){
+  var coordsToDraw = this.createLatLongObjects(geoJsonObject)
+
+    bufferedPolygon = new google.maps.Polygon({
+    paths: coordsToDraw,
+    strokeColor: '#FF0000',
+    strokeOpacity: 0.8,
+    strokeWeight: 2,
+    fillColor: '#FF0000',
+    fillOpacity: 0.35
+  });
+  bufferedPolygon.setMap(map);
+
+}
+
+//------------Markers----------------
+
+function loadMarkers(attractions){
+    $.each(attractions, function(i,item){
+      loadMarker(item);
+    });
+  }
+
+function loadMarker(attraction){
+
+  var latitude = attraction.longlat.coordinates[1];
+  var longitude = attraction.longlat.coordinates[0];
+  var coords = new google.maps.LatLng(latitude, longitude);
+
+  var marker = new google.maps.Marker({
+    position: coords,
+    map: map
+  });
+
+  new InfoBox(attraction, marker)
+  loadAttractionList(attraction, marker)
+}
+
+//-----------InfoBox----------------
+var InfoBox = function(attraction, marker){
+  this.contentString ='<div>' +
```

Comment:
Generally, it's better practice to only store needed variables (attraction, marker) in the constructor function and then add an `init` method which does set up.  This prevents you from copying a lot of code between each object.  Or if this is a one-time only object, you might consider using an object literal.  

What do you think @m3rabb?